### PR TITLE
feat: user profiles api

### PR DESCRIPTION
**Pull request by danlevings**
Introduce the user profiles API

### CODE REVIEW

File: packages/stable/src/api/userProfiles/profilesApi.ts
Comment (andersfylling) [inline packages/stable/src/api/userProfiles/profilesApi.ts]: is `UserProfilesListResponse` a better fit?

Comment (danlevings) [reply packages/stable/src/api/userProfiles/profilesApi.ts]: yes! thanks!

**Comment by andersfylling** [comment]
I would recommend using the generated request and response objects in your endpoint implementations. Then the endpoints are also updated whenever your request/response objects are updated as well.

File: packages/stable/src/__tests__/api/userProfiles.nit.spec.ts
Comment (stefanmaretic) [inline packages/stable/src/__tests__/api/userProfiles.nit.spec.ts]: I guess this should be `profiles`?

File: packages/stable/src/__tests__/api/userProfiles.nit.spec.ts
Comment (stefanmaretic) [inline packages/stable/src/__tests__/api/userProfiles.nit.spec.ts]: Should we test here for `expect(nextProfiles).not.toEqual(profiles)`?

File: packages/stable/src/__tests__/api/userProfiles.nit.spec.ts
Comment (stefanmaretic) [inline packages/stable/src/__tests__/api/userProfiles.nit.spec.ts]: Is this just the exact same test as the one above?

File: packages/stable/src/__tests__/api/userProfiles.nit.spec.ts
Comment (stefanmaretic) [inline packages/stable/src/__tests__/api/userProfiles.nit.spec.ts]:
```suggestion
```

File: packages/stable/src/api/userProfiles/profilesApi.ts
Comment (stefanmaretic) [inline packages/stable/src/api/userProfiles/profilesApi.ts]: Not sure whether the year is relevant here, but 2023? 😬

**Review by stefanmaretic — COMMENTED** [review]
I don't think any of my comments are blockers, but would be nice to resolve them before merging.

File: packages/stable/src/__tests__/api/userProfiles.nit.spec.ts
Comment (gulkaktus) [reply packages/stable/src/__tests__/api/userProfiles.nit.spec.ts]: fixed!

File: packages/stable/src/__tests__/api/userProfiles.nit.spec.ts
Comment (gulkaktus) [reply packages/stable/src/__tests__/api/userProfiles.nit.spec.ts]: fixed!

File: packages/stable/src/__tests__/api/userProfiles.nit.spec.ts
Comment (gulkaktus) [reply packages/stable/src/__tests__/api/userProfiles.nit.spec.ts]: removed!

File: packages/stable/src/__tests__/api/userProfiles.nit.spec.ts
Comment (gulkaktus) [reply packages/stable/src/__tests__/api/userProfiles.nit.spec.ts]: Removed

File: packages/stable/src/api/userProfiles/profilesApi.ts
Comment (gulkaktus) [reply packages/stable/src/api/userProfiles/profilesApi.ts]: fixed!

**Comment by gulkaktus** [comment]
Tests are failing because user ids are currently not generated for service accounts. This is expected to be fixed around October 1st - 5th

File: packages/stable/src/api/userProfiles/profilesApi.ts
Comment (haakonvt) [inline packages/stable/src/api/userProfiles/profilesApi.ts]: I dont read js, but just a heads up that byids doesn't return in order. Here's how the Python SDK manually sorts results before returning:
https://github.com/cognitedata/cognite-sdk-python/blob/f4c00fe6fcae3fc9d821e3e9ca5444ac0e11d9a9/cognite/client/_api/user_profiles.py#L81

Comment (gulkaktus) [reply packages/stable/src/api/userProfiles/profilesApi.ts]: Thanks for the heads up! Should probably be updated at some point, but I'll leave this as is since it's how we're currently treating all the retrieve methods.

File: packages/stable/src/__tests__/api/userProfiles.nit.spec.ts
Comment (polomani) [inline packages/stable/src/__tests__/api/userProfiles.nit.spec.ts:1]: .unit.spec.ts (misspelled)

Comment (gulkaktus) [reply packages/stable/src/__tests__/api/userProfiles.nit.spec.ts:1]: good catch, fixed!

File: packages/stable/src/__tests__/api/userProfiles.unit.spec.ts
Comment (polomani) [inline packages/stable/src/__tests__/api/userProfiles.unit.spec.ts:1]: looking at this again, those are integration tests, not unit, right?

File: packages/stable/src/api/userProfiles/profilesApi.ts
Comment (polomani) [inline packages/stable/src/api/userProfiles/profilesApi.ts]: is there a chance to make better links for this, e.g. by giving an operation name?
e.g. sessions: `https://api-docs.cognite.com/20230101/tag/Sessions/operation/listSessions`
compare to `https://api-docs.cognite.com/20230101/tag/User-profiles/paths/~1profiles~1search/post`

Comment (polomani) [reply packages/stable/src/api/userProfiles/profilesApi.ts]: the reason I ask, is that they won't appear as code snippets in the portal otherwise, example:

Screenshot 2023-11-28 at 16 08 15

Comment (gulkaktus) [reply packages/stable/src/api/userProfiles/profilesApi.ts]: Good point, I guess I should modify this file? https://github.com/cognitedata/service-contracts/blob/master/versions/v1/user_profiles.yml

The current link is being referenced by others, e.g. the Python SDK. What's the best way of communicating to everyone that the link will break? @polomani

File: packages/stable/src/__tests__/api/userProfiles.unit.spec.ts
Comment (gulkaktus) [reply packages/stable/src/__tests__/api/userProfiles.unit.spec.ts:1]: Yes, fixed!

File: packages/stable/src/api/userProfiles/profilesApi.ts
Comment (polomani) [reply packages/stable/src/api/userProfiles/profilesApi.ts]: it might not break, as there is a chance that both will work no? All other resources use operationids anyway, it's better to be consistent.
Also it should be possible to update the link in python sdk

Comment (gulkaktus) [reply packages/stable/src/api/userProfiles/profilesApi.ts]: Made a PR: https://github.com/cognitedata/service-contracts/pull/2388
Awaiting review from the API team

Comment (gulkaktus) [reply packages/stable/src/api/userProfiles/profilesApi.ts]: @polomani above PR merged and urls updated 😃

File: packages/stable/src/api/userProfiles/profilesApi.ts
Comment (polomani) [inline packages/stable/src/api/userProfiles/profilesApi.ts]: you could reuse the common type
```suggestion
  public list = (scope?: FilterQuery): UserProfilesListResponse => {
    return this.listEndpoint(this.callListEndpointWithGet, scope);
  };
```

File: packages/stable/src/api/userProfiles/profilesApi.ts
Comment (polomani) [inline packages/stable/src/api/userProfiles/profilesApi.ts]: this snippet is incorrect
Screenshot 2023-11-30 at 16 28 59

File: packages/stable/src/api/userProfiles/profilesApi.ts
Comment (gulkaktus) [reply packages/stable/src/api/userProfiles/profilesApi.ts]: ah nice

File: packages/stable/src/api/userProfiles/profilesApi.ts
Comment (gulkaktus) [reply packages/stable/src/api/userProfiles/profilesApi.ts]: Fixed!

File: packages/stable/src/__tests__/api/userProfiles.int.spec.ts
Comment (polomani) [inline packages/stable/src/__tests__/api/userProfiles.int.spec.ts]:
```suggestion
describe('User Profiles', () => {
```

File: packages/stable/src/exports.gen.ts
Comment (polomani) [inline packages/stable/src/exports.gen.ts:130]: it seems like some of these types aren't needed/used, @andersfylling is there a setting to omit them somehow?

Comment (andersfylling) [reply packages/stable/src/exports.gen.ts:130]: Hm, I don't think we added support for that. Question here is why aren't they using an ErrorResponse type in the open api spec?

**Review by polomani — APPROVED** [review]
LGTM, please squash when merging (and keep "feat: user profiles api" as the commit name)

**Pull request merged**
